src/main/java/com/faforever/client/replay/ReplayServiceImpl.java

```diff
@@ -161,7 +161,7 @@ static String guessModByFileName(String fileName) {
       for (Path replayFile : directoryStream) {
         try {
           LocalReplayInfo replayInfo = replayFileReader.parseMetaData(replayFile);
-          FeaturedMod featuredMod = modService.getFeaturedMod(replayInfo.getFeaturedMod()).getNow(FeaturedMod.UNKNOWN);
+          FeaturedMod featuredMod = modService.getFeaturedMod(replayInfo.getFeaturedMod()).get();
```

Comment:
getNow is shit as this is connected to a api call, we could also use a get with timeout, but the api call has a timeout it self

Reply:
We shouldn't use getNow() nor get(), but instead use thenAccept(), right?

Reply:
Yeah we should

Reply:
No we can but we don't need to it is all inside a CompletedFutureTask, so I have to call get at some point anyway ;)